core/src/main/java/org/apache/calcite/sql/SqlCreateProcedure.java

```diff
@@ -105,7 +105,12 @@ public SqlCreateProcedure(SqlParserPos pos, SqlCreateSpecifier createSpecifier,
 
   @Override public void validate(SqlValidator validator,
       SqlValidatorScope scope) {
-    if (statement instanceof SqlScriptingNode) {
+    if (statement instanceof SqlScriptingNode
```

Comment:
Since this logic looks identical to validateSqlNodeList in SqlScriptingNode, should we move it to a common location (e.g. SqlValidatorImpl) and call it from both locations?

Reply:
Yeah that sounds good. I'll add another method to move this logic out